akka-projection-kafka/src/main/scala/akka/projection/kafka/internal/KafkaSourceProviderImpl.scala

```diff
@@ -92,15 +95,16 @@ import org.apache.kafka.common.record.TimestampType
     }
   }
 
-  override def source(readOffsets: Supplier[CompletionStage[Optional[GroupOffsets]]])
+  override def source(readOffsets: Supplier[CompletionStage[Optional[MergeableOffset[JLong]]]])
       : CompletionStage[akka.stream.javadsl.Source[ConsumerRecord[K, V], NotUsed]] = {
     source(() => readOffsets.get().toScala.map(_.asScala)).map(_.asJava).toJava
   }
 
-  override def extractOffset(record: ConsumerRecord[K, V]): GroupOffsets = GroupOffsets(record)
+  override def extractOffset(record: ConsumerRecord[K, V]): MergeableOffset[JLong] =
+    KafkaOffsets.toMergeableOffset(record)
 
-  override def verifyOffset(offsets: GroupOffsets): OffsetVerification = {
-    if (offsets.partitions.forall(assignedPartitions.contains))
```

Comment:
Ok, I see. 

I would prefer to merge this PR because it fixes a blocking bug. We can revisit it and try to optimise after `v1.0.0`. Or during RC-1 phase.

Reply:
Create an issue for it, so we don't lose it from sight.
https://github.com/akka/akka-projection/issues/328